engine/v2/transaction/beef/beef_service_test.go

```diff
@@ -0,0 +1,272 @@
+package beef_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/bitcoin-sv/go-sdk/chainhash"
+	sdk "github.com/bitcoin-sv/go-sdk/transaction"
+	"github.com/bitcoin-sv/spv-wallet/engine/v2/transaction/beef"
+	"github.com/bitcoin-sv/spv-wallet/engine/v2/transaction/beef/testabilities"
+	txerrors "github.com/bitcoin-sv/spv-wallet/engine/v2/transaction/errors"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPrepareBEEF_MissingSourceTx_InSubjectTxInput(t *testing.T) {
```

Comment:
Maybe let's add more edge cases to those tests (I would like to cover all paths in tests).
For example:
-> when source ID from input is invalid hex,
-> when BEEF has invalid hex format,
-> malformed transaction structure in PrepareBEEF